packages/relay/src/lib/eth.ts

```diff
@@ -704,17 +704,29 @@ export class EthImpl implements Eth {
   /**
    * Perform value format precheck before making contract call towards the mirror node
    * @param transaction
+   * @param requestIdPrefix
    */
-  contractCallFormat(transaction: IContractCallRequest): void {
+  async contractCallFormat(transaction: IContractCallRequest, requestIdPrefix?: string): Promise<void> {
     if (transaction.value) {
       transaction.value = weibarHexToTinyBarInt(transaction.value);
     }
     if (transaction.gasPrice) {
       transaction.gasPrice = parseInt(transaction.gasPrice.toString());
+    } else {
+      transaction.gasPrice = await this.gasPrice(requestIdPrefix).then((gasPrice) => parseInt(gasPrice));
     }
     if (transaction.gas) {
       transaction.gas = parseInt(transaction.gas.toString());
     }
+    if (!transaction.from && transaction.value && transaction.value > 0) {
```

Comment:
Nit.  The warning "Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly." applies here.  I know it also applies to the other conditions, but slowly cleaning up code as we work on it, is a good practice.

Reply:
Actually, this warning only applies if you pass the `string` directly into the conditional. Using `!` or `!!` converts the `string` to a `boolean`, which avoids the issue. We can address other conditions over time, but this one is already handled correctly.